Integrate LSP4Jakarta 0.2.2 feature branch into main.

mrglavas opened the pull request:
Resolves https://github.com/OpenLiberty/liberty-tools-intellij/issues/1077

Note that an upgrade to LSP4MP 0.13.0 was also included in the initial PR https://github.com/OpenLiberty/liberty-tools-intellij/pull/1061 that was merged to the feature branch.

mrglavas commented on .github/workflows/build.yaml:
```diff
@@ -42,7 +42,7 @@ on:
     branches: '**'
   pull_request:
-    branches: [ main ]
+    branches: [ main, lsp4jakarta-0.2.2-integration ]
```
```suggestion
    branches: [ main ]
```

turkeylurkey reviewed (approved):
+1

TrevCraw reviewed (approved):
Might be good to mention LSP4MP is also updated as part of this PR in the description.

mrglavas commented:
> Might be good to mention LSP4MP is also updated as part of this PR in the description. Generally, Windows builds have been passing more in main. Do you know why these Windows builds failed?

@TrevCraw I only see one failure in the Windows PR build and it's a timeout on a server.env diagnostic test. Doesn't sound like a change to LSP4Jakarta could have caused this.

```
GradleSingleModLSTest > testDiagnosticInServerEnv() FAILED
    java.lang.RuntimeException: Hover on text: NONE did not trigger a pop-up window to open
        at io.openliberty.tools.intellij.it.UIBotTestUtils.hoverInAppServerCfgFile(UIBotTestUtils.java:859)
        at io.openliberty.tools.intellij.it.SingleModLibertyLSTestCommon.testDiagnosticInServerEnv(SingleModLibertyLSTestCommon.java:350)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
        at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
        at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
        at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:156)
        at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:147)
        at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:86)
```

mrglavas commented:
> Might be good to mention LSP4MP is also updated as part of this PR in the description. Generally, Windows builds have been passing more in main. Do you know why these Windows builds failed?

@TrevCraw The upgrade to LSP4MP is mentioned in the the first PR on the feature branch, but I added a note on the description so someone can see that without having to browse that far through links.
